src/webview/markdownEngine.ts

```diff
@@ -0,0 +1,99 @@
+import * as hljs from "highlight.js";
+import * as MarkdownIt from "markdown-it";
+import * as path from "path";
+import * as vscode from "vscode";
+import { leetCodeChannel } from "../leetCodeChannel";
+
+export class MarkdownEngine {
+
+    public readonly engine: MarkdownIt;
```

Comment:
private

Reply:
Switch to this routine:
```typescript
private readonly _engine: MarkdownIt;
public get engine() { return this.engine; }
```
?

Reply:
If we adopt the workaround (`host` and `defaultLang`) above, maybe there is no need to expose engine and extPath anymore.